### PR TITLE
fix(gatsby-source-contentful): Restricts images to 4000 max image width/height

### PR DESCRIPTION
**Pull request by bigfanjs**
## Description
Since Contentful does not allow fetching images above `4000` in size, this PR restricts images to not go above that maximum size to prevent images from not even loading in large screens.

## Related Issues
Related to #14752

### CODE REVIEW

File: packages/gatsby-source-contentful/src/extend-node-type.js [hunk 1]
```diff
@@ -114,7 +116,9 @@ const resolveFixed = (image, options) => {
   fixedSizes = fixedSizes.map(Math.round)
 
   // Filter out sizes larger than the image's width.
-  const filteredSizes = fixedSizes.filter(size => size <= width)
+  const filteredSizes = fixedSizes.filter(
+    size => size <= contentful_max_size || size <= width
```
Comment (wardpeet) [inline packages/gatsby-source-contentful/src/extend-node-type.js]: I've been testing this condition and I think you want `&&` as we want to filter out images that are bigger than 4000 or bigger than the image width. By changing this the test will fail but that's a good thing :wink: 

```suggestion
    size => size <= contentful_max_size && size <= width
```

File: packages/gatsby-source-contentful/src/extend-node-type.js [hunk 2]
```diff
@@ -215,7 +219,9 @@ const resolveFluid = (image, options) => {
   fluidSizes = fluidSizes.map(Math.round)
 
   // Filter out sizes larger than the image's maxWidth.
-  const filteredSizes = fluidSizes.filter(size => size <= width)
+  const filteredSizes = fluidSizes.filter(
+    size => size <= contentful_max_size || size <= width
```
Comment (wardpeet) [inline packages/gatsby-source-contentful/src/extend-node-type.js]: This line has the same issue as above but there is a catch 😂

A few lines below we have the following line of code
https://github.com/gatsbyjs/gatsby/blob/ca4f15af9a68302a78436af60801e973b9907416/packages/gatsby-source-contentful/src/extend-node-type.js#L228

This one fights with the contentful_max_size constant. If the image is bigger than contentful_max_size we still add it.

So I suggest changing that line to
`if (!filteredSizes.includes(parseInt(width)) && parseInt(width) < contentful_max_size) filteredSizes.push(width)`

File: packages/gatsby-source-contentful/src/extend-node-type.js [hunk 3]
```diff
@@ -20,6 +20,8 @@ const {
   ImageCropFocusType,
 } = require(`./schemes`)
 
+const contentful_max_size = 4000
```
[the remark this answers is not in the document]
Comment (bigfanjs) [reply packages/gatsby-source-contentful/src/extend-node-type.js]: Yeah sure I like this too.

File: packages/gatsby-source-contentful/src/extend-node-type.js (hunk 1 above)
Comment (bigfanjs) [reply packages/gatsby-source-contentful/src/extend-node-type.js]: Oops! What a mistake!

File: packages/gatsby-source-contentful/src/extend-node-type.js (hunk 2 above)
Comment (bigfanjs) [reply packages/gatsby-source-contentful/src/extend-node-type.js]: aah nice!!

**Comment by sidharthachatterjee** [comment]
Published in `gatsby-source-contentful@2.0.72`

**Pull request merged**